tapiriik/services/Setio/setio.py

```diff
@@ -0,0 +1,286 @@
+#
+#   Created by Christian Toft Andersen 2017 for SETIO @
+#
+from tapiriik.settings import WEB_ROOT, SETIO_CLIENT_SECRET, SETIO_CLIENT_ID
+from tapiriik.services.service_base import ServiceAuthenticationType, ServiceBase
+from tapiriik.services.service_record import ServiceRecord
+from tapiriik.database import cachedb
+from tapiriik.services.interchange import UploadedActivity, ActivityType, ActivityStatistic, ActivityStatisticUnit, Waypoint, WaypointType, Location, Lap
+from tapiriik.services.api import APIException, UserException, UserExceptionType, APIExcludeActivity
+from tapiriik.services.fit import FITIO
+from datetime import datetime, timedelta
+
+from django.core.urlresolvers import reverse
+from datetime import datetime, timedelta
+from urllib.parse import urlencode
+import calendar
+import requests
+import os
+import logging
+import pytz
+import re
+import time
+import json
+import dateutil.parser
+
+logger = logging.getLogger(__name__)
+
+class SetioService(ServiceBase):
+    ID = "setio"
+    DisplayName = "Setio"
+    DisplayAbbreviation = "SET"
+    AuthenticationType = ServiceAuthenticationType.OAuth
+    AuthenticationNoFrame = True  # They don't prevent the iframe, it just looks really ugly.
+    PartialSyncRequiresTrigger = True

```

Comment:
Does the site actually send webhook notifications upon new activities? If not, this should be false (and you can delete the sub/unsub functions later on, too). If so, there needs to be a function to receive the webhook and return user IDs, see Strava or Endomondo as examples.